ec/ecresource/deploymentresource/deployment/v2/schema.go

```diff
@@ -132,3 +141,38 @@ func DeploymentSchema() tfsdk.Schema {
 		},
 	}
 }
+
+type setUnknownIfResetPasswordIsTrue struct{}
+
+var _ tfsdk.AttributePlanModifier = setUnknownIfResetPasswordIsTrue{}
+
+func (m setUnknownIfResetPasswordIsTrue) Description(ctx context.Context) string {
+	return m.MarkdownDescription(ctx)
+}
+
+func (m setUnknownIfResetPasswordIsTrue) MarkdownDescription(ctx context.Context) string {
+	return "Sets the planned value to unknown if the reset_elasticsearch_password config value is true"
+}
+
+func (m setUnknownIfResetPasswordIsTrue) Modify(ctx context.Context, req tfsdk.ModifyAttributePlanRequest, resp *tfsdk.ModifyAttributePlanResponse) {
+	if resp.AttributePlan == nil || req.AttributeConfig == nil {
+		return
+	}
+
+	// if the config is the unknown value, use the unknown value otherwise, interpolation gets messed up
+	if req.AttributeConfig.IsUnknown() {
+		return
+	}
+
+	var isResetting *bool
+	resp.Diagnostics.Append(req.Plan.GetAttribute(ctx, path.Root("reset_elasticsearch_password"), &isResetting)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	if isResetting != nil && *isResetting {
+		planVal := resp.AttributePlan.(types.String)
+		planVal.Unknown = true
+		resp.AttributePlan = planVal
```

Comment:
```suggestion
		resp.AttributePlan = types.String{Unknown: true}
```

Reply:
Sorry, the suggestion doesn't reflect the all proposed changes. `planVal` can be totally removed.

Reply:
All good. Good pickup, I was testing something out here and forgot to revert it.